component/http/middleware_test.go

```diff
@@ -688,3 +688,45 @@ func TestIsConnectionReset(t *testing.T) {
 		})
 	}
 }
+
+type failWriter struct {
```

Comment:
@mantzas wasn't sure how you prefer mocking, but I didn't see any mock package, so I created a `failWriter` implementation here, let me know if this needs additional adjustments.

Reply:
Perfect, thanks